src/App.tsx

```diff
@@ -26,70 +27,121 @@ type ReorderOptions = {
+
+      default:
+        return 0;
+    }
+  });
+
+  return isReversed ? visibleGoods.reverse() : visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export class App extends React.Component<{}, State> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  render() {
+    const {
+      sortType,
+      isReversed,
+    } = this.state;
+    const goods = getReorderedGoods(goodsFromServer, this.state);
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={cn(
+              'section__button',
+              'section__button--alphabet',
+              'is-info',
+              { 'is-light generalButton': sortType !== SortType.ALPHABET },
+            )}
+            onClick={() => {
+              this.setState({ sortType: SortType.ALPHABET });
+            }}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={cn(
+              'section__button',
+              'section__button--length',
+              'is-success',
+              { 'is-light generalButton': sortType !== SortType.LENGTH },
+            )}
+            onClick={() => {
+              this.setState({ sortType: SortType.LENGTH });
+            }}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={cn(
+              'section__button',
+              'section__button--reverse',
+              'is-warning',
+              { 'is-light generalButton': !isReversed },
+            )}
+            onClick={() => {
+              this.setState({ isReversed: !isReversed });
+            }}
+          >
+            Reverse
+          </button>
+
+          {(sortType !== SortType.NONE || isReversed) && (
+            <button
+              type="button"
+              className="
+              section__button
+              section__button--reset
+              is-danger
+              is-light
+              "
```

Comment:
not bad, but don't forget about indentation of the child elements
```suggestion
              className="
                section__button
                section__button--reset
                is-danger
                is-light
              "
```